### PR TITLE
Feature/gs 310 prepare fetch assay data in facile analisis

### PR DESCRIPTION
**Pull request by zmar**
'fetch_assay_data' moved to FacileAnalysis

### CODE REVIEW

File: R/assay-data.R [hunk 1]
```diff
@@ -1,129 +1,82 @@
...
+    ncol(vals) == nrow(sample.info),
+    all(colnames(vals) == sample.info$samid),
+    is.character(feature.info$assay_type),
+    length(unique(feature.info$assay_type)) == 1L,
+    is.numeric(sample.info$libsize), is.numeric(sample.info$normfactor))
+  atype <- feature.info$assay_type[1L]
+  libsize <- sample.info$libsize * sample.info$normfactor
+  if (atype == 'rnaseq') {
+    # we assume these are units that are at the count level
+    out <- edgeR::cpm(vals, libsize, log=log, prior.count=prior.count)
```
Comment (gladkia) [inline R/assay-data.R]: @importFrom ?

File: R/assay-data.R [hunk 2]
```diff
@@ -1,129 +1,82 @@
...
+  
   finfo <- assay_feature_info(x, assay_name, feature_ids=feature_ids) %>%
     collect(n=Inf) %>%
     arrange(hdf5_index)
   atype <- finfo$assay_type[1L]
   ftype <- finfo$feature_type[1L]
   sinfo <- assay_sample_info(x, assay_name, samples) %>%
     mutate(samid=paste(dataset, sample_id, sep="__"))
-  bad.samples <- is.na(sinfo$hdf5_index)
+  bad.samples <- is.na(pull(sinfo, hdf5_index))
```
Comment (gladkia) [inline R/assay-data.R]: Add comment for this 'pull'

**Review by gladkia — CHANGES_REQUESTED** [review]

**Review by phaverty — COMMENTED** [review]
Don't we need the S4 method for organism?

File: R/assay-data.R (hunk 2 above)
Comment (lianos) [reply R/assay-data.R]: Hah! I learned a new `dplyr` verb today, nice thanks! :-)

File: R/assay-data.R (hunk 1 above)
Comment (zmar) [reply R/assay-data.R]: Done

File: R/assay-data.R (hunk 2 above)
Comment (zmar) [reply R/assay-data.R]: I turns out that `pull` is unnecessary so I restored is.na(sinfo$hdf5_index). `dplyr::pull` allows you to pull out a single variable from objects like tbl_dbi.

**Pull request merged**